src/analytics.js

```diff
@@ -804,11 +762,86 @@ window.Altis.Analytics.registerAttribute = ( name, value ) => {
  * @param {string} name The metric name.
  * @param {*} value The metric value, can be a number, or callback or Promise that returns a number.
  */
-window.Altis.Analytics.registerMetric = ( name, value ) => {
+Altis.Analytics.registerMetric = ( name, value ) => {
 	_metrics[ name ] = value;
 	Analytics.updateAudiences();
 };
 
-// Fire a ready event once userland API has been exported.
-const readyEvent = new CustomEvent( 'altis.analytics.ready' );
-window.dispatchEvent( readyEvent );
+/**
+ * Start recording default events and trigger onReady event.
+ */
+function startAnalytics() {
+	// Avoid re-running everything.
+	if ( Altis.Analytics.Ready ) {
+		return;
+	}
+
+	// Set initial endpoint data.
+	Analytics.mergeEndpointData( Data.Endpoint || {} );
+
+	// Track sessions.
+	document.addEventListener( 'visibilitychange', () => {
+		if ( document.hidden ) {
+			// On hide increment elapsed time.
+			elapsed += Date.now() - start;
+			// Fire session stop event.
+			Analytics.record( '_session.stop' );
+		} else {
+			// On show reset start time.
+			start = Date.now();
+			// Reset subSessions.
+			subSessionId = uuid();
+			subSessionStart = Date.now();
+			// Fire session start event.
+			Analytics.record( '_session.start' );
+		}
+	} );
+
+	/**
+	 * Record the default page view.
+	 */
+	const recordPageView = () => {
+		// Session start.
+		Analytics.record( '_session.start' );
+		// Record page view event & create/update endpoint immediately.
+		Analytics.record( 'pageView', false );
+	};
+
+	if ( document.readyState === 'interactive' || document.readyState === 'complete' || document.readyState === 'loaded' ) {
+		recordPageView();
+	} else {
+		window.addEventListener( 'DOMContentLoaded', recordPageView );
+	}
+
+	// Flush remaining events.
+	window.addEventListener( 'beforeunload', () => {
+		Analytics.record( '_session.stop', false );
+	} );
+
+	// Fire a ready event once userland API has been exported.
+	Altis.Analytics.Ready = true;
+	const readyEvent = new CustomEvent( 'altis.analytics.ready' );
+	window.dispatchEvent( readyEvent );
+}
+
+// Check if this is anonymous or not by looking for endpoint user ID.
+const isAnonymous = ! ( Data.Endpoint && Data.Endpoint.User && Data.Endpoint.User.UserId );
+const consentType = 'statistics' + ( isAnonymous ? '-anonymous' : '' );
```

Comment:
This seems to imply an either/or relationship with statistics cookies, e.g. you can't be both looking at general statistics _and_ anonymous statistics. Is this accurate and is that intentional?

Reply:
Hm, so I opted to have this switch in place because if the endpoint `UserId` is set that allows the site to link a logged in users endpoints together and to identify them. Without that there's no identifying data tracked by default hence the `isAnonymous` switch. Maybe we need to allow the category to be filtered on the backend instead to provide control and avoid the need for this logic.

Reply:
I changed it so the default is just `statistics` to keep things simple. You know this stuff better regarding the categories, what do you think would make sense here?

I'm a bit confused what you mean by looking at both non-anonymous and anonymous statistics. From a backend / dashboard perspective you certainly could, but this is just about checking for a suitable category based on the data we're tracking.

Thinking about this a bit further are you suggesting we could do some tracking of data if `statistics-anonymous` is allowed but remove / redact some info until `statistics` is consented to as well?

Reply:
I guess my thought is that `statistics`/`statistics-anonymous` aren't mutually exclusive. I would assume the default to be `statistics-anonymous` -- we're collecting data, but it's not user-identifiable -- and as such, that's allowed through our consent plugin by default. But if we're collecting data that's for the purpose of analytics that _does_ have user-identifying information, then we need permission. 

The check here makes sense, but the two categories are separate. A site can be allowing both anonymous statistics and non-anonymous statistics, or neither and if, in our context, it's always an either/or thing, this is fine.

Reply:
Thanks for the feedback and insight - I've updated it so that anon consent allows tracking but until full stats consent is given the `User` property of the endpoint data is stripped off.